docs/how-tos/rust_service.md

```diff
@@ -0,0 +1,334 @@
+# Rust Service
+
+## Preface
+
+This guide will explain how to develop a Rust service to serve data to the Graph.
+We will cover:
+
+- Setting up the HTTP/GraphQL Server in [Serving GraphQL Requests](#serving-graphql-requests)
+- Creating GraphQL Objects and providing field resolvers in [Defining Objects](#defining-objects)
+- Extending the graph using [Federation](#federated)
+- Exporting the GraphQL schema for use in federation in [Schema Generation](#schema-generation)
+
+## Dependencies
+
+This guide will utilize the following dependencies:
+
+- `axum` as the HTTP server
+- `async-graphql` as graphql server library 
+- `async-graphql-axum` for GraphQL request handling
+- `tokio` as runtime for writing asynchronous rust program
+
+```toml
+[dependencies]
+async-graphql = "7.0.6"
+async-graphql-axum = "7.0.6"
+axum = "0.7.5"
+tokio = { version = "1.38.0", features = ["macros", "rt-multi-thread"] }
+```
+
+## Serving GraphQL Requests
+
+To handle GraphQL requests, we start by creating a server with a placeholder GraphQL schema. We use the `async-graphql-axum` crate, which provides integration between `async-graphql` and `axum`, to create a `graphql_handler`.
+
+!!! example "GraphQL Request Handler"
+
+    ```rust
+    use async_graphql::{EmptyMutation, EmptySubscription, Object, Schema};
+    use async_graphql_axum::{GraphQLRequest, GraphQLResponse};
+    use axum::{
+        extract::Extension,
+        routing::post,
+        Router,
+    };
+
+    struct Query;
+
+    #[tokio::main]
+    async fn main() {
+        let schema = Schema::build(Query, EmptyMutation, EmptySubscription).finish();
+
+        let app = Router::new()
+            .route("/graphql", post(graphql_handler))
+            .layer(Extension(schema));
+
+        let listener = tokio::net::TcpListener::bind("127.0.0.1:8000").await.unwrap();
+        axum::serve(listener, app).await.unwrap();
+    }
+
+    async fn graphql_handler(
+        schema: Extension<Schema<Query, EmptyMutation, EmptySubscription>>,
+        req: GraphQLRequest,
+    ) -> GraphQLResponse {
+        schema.execute(req.into_inner()).await.into()
+    }
+    ```
+
+!!! info
+
+    `GraphQLRequest` handles incoming GraphQL requests. It wraps the raw HTTP request
+    data and converts it into a form that can be processed by async-graphql.
+
+    `GraphQLResponse` wraps the results of executing a GraphQL query. It ensures
+    that the response is correctly formatted as a valid HTTP response that Axum can 
+    send back to the client.
```

Comment:
I would probably include this in the main body of text above as it doesn't need to draw the users attention